ultravox/training/train.py

```diff
@@ -310,17 +311,27 @@ def train(args: config_base.TrainConfig):
         logging.info("Starting training...")
         t_start = datetime.now()
         logging.info(f"train start time: {t_start}")
+
         if args.val_steps:
-            trainer.evaluate()
+            if args.use_fsdp:
+                logging.warning(
+                    "FSDP is enabled: Skipping initial validation since model is not initialized."
+                )
+            else:
+                trainer.evaluate()
+
         trainer.train()
         t_end = datetime.now()
         logging.info(f"train end time: {t_end}")
         logging.info(f"elapsed: {t_end - t_start}")
 
+    if args.use_fsdp:
```

Comment:
From what I understand, we actually don't want `FULL_STATE_DICT` for model checkpoints that are created during training. That's why we set it to `SHARDED_STATE_DICT` above.
We do need this for doing the final save though.

I can't say this with 100% certainty though, since what people said would happen and what I saw happening in practice was different.

Reply:
OK, this definitely seems like black magic stuff, please add a comment indicating this that we're changing the state dict back to non-sharded so it can be serialized properly.